components/forms/SignupForm.tsx

```diff
@@ -0,0 +1,207 @@
+"use client";
+
+import {
+  Container,
+  Form,
+  Button,
+  FormCheck,
+  Row,
+  Col,
+  Spinner,
+  InputGroup,
+} from "react-bootstrap";
+
+import { submitANForm } from "@/utils/AnApiSubmission";
+import { rubik } from "@/utils/Fonts";
+import FormCheckInput from "react-bootstrap/esm/FormCheckInput";
+import FormCheckLabel from "react-bootstrap/esm/FormCheckLabel";
+import { useRef, useState } from "react";
+import ConstituencyLookup from "./constituencyLookup";
+
+const emailErrorMessage = (code: EmailErrorCode) => {
+  switch (code) {
+    case "EMAIL_INVALID":
+      return "Please add a valid email address.";
+    case "SERVER_ERROR":
+      return "Something went wrong signing you up. Please try again?";
+    default:
+      return "";
+  }
+};
+
+type FormData = {
+  email: string;
+  emailOptIn: boolean;
+};
+
+const initialFormState: FormData = {
+  email: "",
+  emailOptIn: false,
+};
+
+const SignupForm = () => {
+  // TODO: Look into using useFormState in future:
+  // https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#server-side-validation-and-error-handling
+  // Was hitting into this issue: https://github.com/vercel/next.js/issues/55919
+
+  // Variables used to track state of the email opt-in components
+  const [formState, setFormState] = useState<FormData>(initialFormState);
+  const [emailError, setEmailError] = useState<EmailErrorCode | null>(null);
+  const formRef = useRef<HTMLFormElement | null>(null);
+
+  // Variables used to track state of the ConstituencyLookup components
+  const validPostcode = useRef("");
+  const [constituency, setConstituency] = useState<Constituency | null>(null);
+  const [constituencyApiLoading, setConstituencyApiLoading] = useState(false);
+
+  const submitForm = async () => {
+    if (!formState.emailOptIn) {
+      // TODO: Set error message - need to opt-in to subscribe
+    } else if (
+      !formRef.current ||
+      formRef.current.email.validity.typeMismatch
+    ) {
+      // TODO: Set error message - invalid email
+    } else if (!constituency) {
+      // TODO: Set error message - input email / select constituency
+    } else {
+      const anResponse = await submitANForm(
+        formState.email,
+        validPostcode.current,
+        constituency,
+        process.env.NEXT_PUBLIC_AN_POSTCODE_FORM || "",
+        ["stop the tories", "movement forward", "election reminders", "join"],
```

Comment:
This is where you add in extra tags. 
So when there are tick boxes for 'I want to volunteer' and 'I'm a student' it should set them as strings in this array.